src/path.rs

```diff
@@ -0,0 +1,790 @@
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let local_data_dir_path = local_data_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn local_data_dir() -> crate::Result<PathBuf> {
+    let raw = inner::localDataDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the user's picture directory.
+///
+/// #### Platform-specific
+///
+/// - **Linux:** Resolves to [`xdg-user-dirs`](https://www.freedesktop.org/wiki/Software/xdg-user-dirs/)' `XDG_PICTURES_DIR`.
+/// - **macOS:** Resolves to `$HOME/Pictures`.
+/// - **Windows:** Resolves to `{FOLDERID_Pictures}`.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::picture_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let picture_dir_path = picture_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn picture_dir() -> crate::Result<PathBuf> {
+    let raw = inner::pictureDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the user's public directory.
+///
+/// #### Platform-specific
+///
+/// - **Linux:** Resolves to [`xdg-user-dirs`](https://www.freedesktop.org/wiki/Software/xdg-user-dirs/)' `XDG_PUBLICSHARE_DIR`.
+/// - **macOS:** Resolves to `$HOME/Public`.
+/// - **Windows:** Resolves to `{FOLDERID_Public}`.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::public_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let public_dir_path = public_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn public_dir() -> crate::Result<PathBuf> {
+    let raw = inner::publicDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the application's resource directory.
+///
+/// To resolve a resource path, see the [`resolve_resource`] function.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::resource_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let resource_dir_path = resource_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn resource_dir() -> crate::Result<PathBuf> {
+    let raw = inner::resourceDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Resolve the path to a resource file.
+///
+/// @param resourcePath The path to the resource.
+/// Must follow the same syntax as defined in `tauri.conf.json > tauri > bundle > resources`, i.e. keeping subfolders and parent dir components (`../`).
+/// @returns The full path to the resource.
+///
+/// ```rust,no_run
+/// use tauri_sys::path::resolve_resource;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let resource_path = resolve_resource("script.sh").await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn resolve_resource(resource_path: &str) -> crate::Result<PathBuf> {
```

Comment:
Should this accept `AsRef<Path>` instead of `&str`?

Reply:
Updated in [7b12c94a](https://github.com/emirror-de/tauri-sys/commit/7b12c94a40bdba63c6c604ae905b9ee018f7cfd2)